### DISCUSSION TITLE
[RFC] Viewer-friendly sorting

### DISCUSSION THREAD

**Discussion by Dum4G** [iptv-org/community#14] [Ideas]
You know, in my country there are several channels that have always been broadcasted for free as long as you have an antenna. Nowadays it's even baked by law that every media streaming service must also broadcast them totally free! No wonders that people got used to watch them most of their time and expect them to be shown first (or at least groupped separately from the rest). Also we have **dozens** of  channels that may only be interesting to a little audience of a single particular town or a corp or a uni. For worse we have more than a few timezones, so if some carrier provides timeshifted broadcasts a single channel can have **up to 11** streams for it! I'm confident that I can reach up to a thousand working streams from sources I know. But I have no doubts that people simply don't get what they expect to see and that's due to poor sorting and not because we can't offer something. That defeats the purpose of using playlists from here, especially if it's used in some primitive player like VLC. Maybe it's worth doing those steps:

To add a category of so-called "Federal channels". If you have those in your country, how do you call them?
To add a category of "Regional" channels, to totally ignore their original category I guess
To make a special category for "Timeshifted" streams and totally ignore their original category
And the most important part is to add an additional ranging (sorting) not only by name but also by the importance of categories: federal first, entertainment second, news third... regional and timeshifted ones come at the end of the list. Arrangement of current groups we have is a debbatable topic. What do you think? @freearhey @BellezaEmporium

**Comment by BellezaEmporium** [comment]
It's interesting to hear about the media landscape in your country and the unique challenges related to channel categorization and sorting. It sounds like you have a diverse range of channels to choose from, and your suggestions for improving the organization of these channels seem quite reasonable.

Regarding your idea to add a category of "Federal channels," as far as I know, these channels may not be present everywhere and might vary in importance or viewership. It's a good idea to distinguish them, as they often play a significant role in providing important national content.

Adding a category for "Regional" channels is a practical approach, as it can help streamline content for a specific audience, and it's not uncommon for regional channels to have a distinct focus.

Creating a separate category for "Timeshifted" streams also makes sense, as this is a unique feature that some viewers may find valuable. It would be helpful to have them organized separately, given the time zone variations.

However, one question tickles my mind. In the case of the United States, that has 3/4 timezones in its own country, and Russia, covering 11 timezones !, if the channels are already split by group/network, how shall we present them on the generated playlists ? Shall we only show the channels, depending on their timezones (-12 to +12), or are we doing those verifications playlist by playlist ? What bothers me is that in the end, we finish with a whole bunch of nonsense, with timezones fighting with each other.

But, to conclude, implementing these steps could indeed enhance the effectiveness of playlists and make it more user-friendly, even for those using simpler media players like VLC.

**Reply by Dum4G** [reply]
We can live it debatable to US active maintainers, 3 to 4 broadcasts is not too much. The Russian way of solving the situation would be to offer Washington (+0 for them?) streams in main groups and the rest of streams in a timeshifted category. This is only applies to generated playlists

**Reply by BellezaEmporium** [reply]
AFAIK Washington is UTC-4.

**Reply by Dum4G** [reply]
As per TV broadcasts we count from Moscow time, meaning the fact that it's GMT+3 is not taken into account. Again, not sure if those rules are applicable to the US

**Comment by freearhey** [comment]
I understand correctly, it is proposed to sort channels by the difference between the time zone of the source and the capital city?

**Reply by BellezaEmporium** [reply]
As far as I remember @Dum4G mentioned that certain channels were made free-to-air on purpose (and it's written by law to provide a free alternative).

Now, those channels need to cover different timezones, which we debated (Russia itself has 11 timezones), and this would mean the addition of a specific category that's either "Timeshifted" or "Federal" channel.

Methinks he had Rossiya 1/24 in mind, or any other Russian public television channel.

I wasn't against, but I thought about merging this into a "regional" category.

**Reply by freearhey** [reply]
And the “regional” category should replace all the others (auto, movies, ...)? Right?

**Reply by Dum4G** [reply]
> I understand correctly, it is proposed to sort channels by the difference between the time zone of the source and the capital city?

1. Federal channels shall be in it's own and the only one category, ranked first in the list. To preserve info about it's genre for possible scenarios I suggest to add isFederal boolean into database
2. In case if Federal channels do have an assigned slot/numeration it shall be stated in the database and be respected by formatter with greater priority than alphabetical order
3. In case if timeshifted streams of any channel exist those shall be placed into Regional category (even if it's for a Federal channel). As an alternative, separate category for timeshifted streams may be proposed
4. Regional channels shall have is it's own and the only one category, ranked at the bottom of playlist in alphabetical order. Regional streams shall broadcast content and/or advertisment oriented for select town, region or timezone and the subdivision must be assigned in the database
5. List of all channels shall also be sorted by categories. Priority is debatable

**Reply by freearhey** [reply]
And which “federal” channels are we talking about (excluding Russian ones)?

**Reply by Dum4G** [reply]
> And which “federal” channels are we talking about (excluding Russian ones)?

https://en.wikipedia.org/wiki/Public_broadcasting

🇬🇧 United Kingdom
In the UK, public broadcasters such as the BBC, ITV, Channel 4, and Channel 5 are required to provide free access to their channels via terrestrial broadcasting. This is ensured by legislation regulating public broadcasting, which sets obligations for broadcasters to provide diverse and quality content accessible to a wide audience without additional fees.

🇩🇪 Germany
Germany has a public broadcasting system that includes organizations like ARD and ZDF. These organizations are required to provide free access to their channels through terrestrial broadcasting. 

🇯🇵 Japan
In Japan, NHK is the sole public broadcaster. All households and companies are required to pay a license fee, which forms the primary funding source for NHK. The law prohibits NHK from airing commercials, which ensures its independence from commercial interests and allows it to provide diverse, quality content without additional charges for viewers.

🇬🇪 Georgia
In Georgia, the broadcasting law states that the public broadcaster must ensure free access to its channels for the population. The law guarantees editorial independence and sets obligations to provide diverse content, including news, educational programs, and children’s programming. The public broadcaster is funded from the state budget, ensuring its independence from commercial interests.

🇮🇸 Iceland
In Iceland, RÚV is the state public broadcaster that provides free access to its channels via terrestrial broadcast. Funding is provided through a special tax collected from each taxpayer, ensuring independence from commercial interests and enabling it to offer diverse, quality content without extra fees for viewers.

🇦🇺 Australia
In Australia, the government has proposed new legislation requiring smart TV manufacturers to give priority access to Australian free-to-air channels such as ABC, SBS, Seven, Nine, and Ten. This legislation aims to ensure easy access to local news and support the local cultural industry. The bill is currently under consideration in parliament.

🇧🇬 Bulgaria
In Bulgaria, the Radio and Television Act guarantees freedom to create radio programs and television broadcasting, prohibits censorship, and ensures freedom of reception and retransmission of programs in the country. The law also establishes principles such as guaranteeing the right to free expression of opinions and the right to information, which promotes free access to diverse content for the population.

**Reply by freearhey** [reply]
Maybe we just add a new "Public" category for these channels?

**Reply by Dum4G** [reply]
Yeah, Public is more common naming for this kind of channels. I believe this category should be compiled by isPublic boolean in case if someone will create a software based on the database where the sorting is different

**Reply by freearhey** [reply]
Well, for other developers sorting by category list should not be a problem, but for us to add a new variable to the database is a lot of work. So for now I think it would be better just to create a regular category.

As for regional channels, the updated [subdivision playlists](https://github.com/iptv-org/iptv/blob/master/PLAYLISTS.md) now include both national channels and channels intended only for this location. So if the data in our database is correct, there should be no extra channels in this playlists.
